### PR TITLE
[acix-24] Notify job status as infra failures using job_failure

### PR DESCRIPTION
**Pull request by CelianR**
### What does this PR do?

Adds checks to set job status as infra failure with job_failure attribute of gitlab jobs.

Related to https://datadoghq.atlassian.net/browse/ACIX-24 but job_failure is already used in some checks.

### Additional Notes

- To be merged after https://github.com/DataDog/datadog-agent/pull/24070 and https://github.com/DataDog/datadog-agent/pull/24666

### CODE REVIEW

File: tasks/libs/pipeline/data.py [hunk 1]
```diff
@@ -120,15 +120,29 @@ def get_failed_jobs(project_name: str, pipeline_id: str) -> FailedJobs:
...
 
 
-def get_job_failure_context(job_log):
+def get_job_failure_context(job: ProjectJob, job_log: str):
     """
     Parses job logs (provided as a string), and returns the type of failure (infra or job) as well
     as the precise reason why the job failed.
     """
 
+    infra_job_failure_reasons = {
```
Comment (chouetz) [inline tasks/libs/pipeline/data.py:129]: I think you should rather store this in the FailedJob types, not to be forced to store this here as well. I guess we probably want to have only one reference for this failure list, wdyt?

File: tasks/unit-tests/pipeline_lib_tests.py [hunk 2]
```diff
@@ -12,3 +17,69 @@ def test_files_loaded_correctly(self):
...
+            attrs={
+                "name": "test",
+                "id": 618,
+                "stage": "test",
+                "status": "failed",
+                "tag_list": [],
+                "allow_failure": False,
+                "web_url": "https://sometest.test",
+                "retry_summary": ["failed"],
+                "failure_reason": "script_failure",
```
Comment (chouetz) [inline tasks/unit-tests/pipeline_lib_tests.py:78]: Do we need a test with an unexpected failure_reason e.g. `"failure_reason": "this_is_an_unexpected_failure_reason"`?

**Review by chouetz — COMMENTED** [review]
Could you check if you can prevent the duplication of the list of failures?
Thanks in advance

File: tasks/unit-tests/pipeline_lib_tests.py (hunk 2 above)
Comment (CelianR) [reply tasks/unit-tests/pipeline_lib_tests.py:78]: As discussed, it is not necessary

**Review by pducolin — COMMENTED** [review]
Leaving the review to @DataDog/agent-ci-experience as it impacts only the CI

**Review by chouetz — APPROVED** [review]

**Pull request merged**